### PR TITLE
chore: use image builder for AKSe VHDs

### PR DESCRIPTION
**Pull request by haofan-ms**
**Reason for Change**:

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [ ] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: temp/AKSeImageTemplateWindows.json [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+                        "vnetConfig": {
+                            "subnetId": "[concat('/subscriptions/',parameters('subscription_id'),'/resourcegroups/',parameters('vnet_resource_group_name'),'/providers/Microsoft.Network/virtualNetworks/',parameters('vnet_name'),'/subnets/',parameters('subnet_name'))]"
+                        }
+                    },
+                "source": {
+                    "type": "PlatformImage",
+                    "publisher": "MicrosoftWindowsServer",
+                    "offer": "WindowsServer",
+                    "sku": "2019-Datacenter-Core-smalldisk",
+                    "version": "17763.5329.231230"
```
Comment (jadarsie) [inline temp/AKSeImageTemplateWindows.json:100]: version should probably be a pipeline variable

File: temp/AKSeImageTemplateWindows.json [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+                        "runElevated": true,
+                        "scriptUri": "[uri('https://raw.githubusercontent.com/Azure/aks-engine-azurestack/',concat(parameters('github_branch_name'),'/vhd/packer/configure-windows-vhd-phase1.ps1'))]"
+                    },
+                    {
+                        "type": "WindowsRestart",
+                        "restartCheckCommand": "echo Azure-Image-Builder-Restarted-the-VM  > c:\\buildArtifacts\\azureImageBuilderRestart.txt",
+                        "restartTimeout": "10m"
+                    },
+                    {
+                        "type": "WindowsRestart",
```
Comment (jadarsie) [inline temp/AKSeImageTemplateWindows.json:115]: do we need 2 restarts?

File: temp/ImageCreationRole.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        "Microsoft.Compute/galleries/images/versions/write",
+
+        "Microsoft.Compute/images/write",
+        "Microsoft.Compute/images/read",
+        "Microsoft.Compute/images/delete"
+    ],
+    "NotActions": [
+    ],
+    "AssignableScopes": [
+      "/subscriptions/a2b58cb8-c2cf-4afe-af64-63ed4dab8628/resourceGroups/AKSeVHDs"
```
Comment (jadarsie) [inline temp/ImageCreationRole.json:18]: can we use variables?

File: temp/ImageTemplateLinux.json [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+    "apiVersion": "2022-02-14",
+    "location": "eastus",
+    "dependsOn": [],
+    "tags": {
+        "userIdentity": "enabled"
+    },
+    "identity": {
+        "type": "UserAssigned",
+        "userAssignedIdentities": {
+            "/subscriptions/a2b58cb8-c2cf-4afe-af64-63ed4dab8628/resourcegroups/AKSeVHDs/providers/Microsoft.ManagedIdentity/userAssignedIdentities/akse-e2e-uai": {}
```
Comment (jadarsie) [inline temp/ImageTemplateLinux.json:12]: please remove hard-coded guids and resource names

File: vhd/packer/install-dependencies.sh [hunk 5]
```diff
@@ -409,13 +409,26 @@ df -h
...
 { printf "Installed apt packages:\n"; apt list --installed | grep -v 'Listing...'; } >> ${VHD_LOGS_FILEPATH}
-{
-  echo "Install completed successfully on " $(date)
-  echo "VSTS Build NUMBER: ${BUILD_NUMBER}"
-  echo "VSTS Build ID: ${BUILD_ID}"
-  echo "Commit: ${COMMIT}"
-  echo "Feature flags: ${FEATURE_FLAGS}"
-} >> ${VHD_LOGS_FILEPATH}
+
+if [ "$#" -eq 4 ]; then
```
Comment (jadarsie) [inline vhd/packer/install-dependencies.sh:413]: is it acceptable a situation where `$#` is not 4?

File: vhd/packer/sysprep.ps1 [hunk 6]
```diff
@@ -27,10 +27,10 @@ $values | ForEach-Object {
 }
 
 # run Sysprep
-if( Test-Path $Env:SystemRoot\\system32\\Sysprep\\unattend.xml ) {  Remove-Item $Env:SystemRoot\\system32\\Sysprep\\unattend.xml -Force }
-& $env:SystemRoot\\System32\\Sysprep\\Sysprep.exe /oobe /generalize /mode:vm /quiet /quit
+#if( Test-Path $Env:SystemRoot\\system32\\Sysprep\\unattend.xml ) {  Remove-Item $Env:SystemRoot\\system32\\Sysprep\\unattend.xml -Force }
```
Comment (jadarsie) [inline vhd/packer/sysprep.ps1:30]: why comment this line? do we want to keep `unattended.xml`?

File: vhd/packer/sysprep.ps1 [hunk 7]
```diff
@@ -27,10 +27,10 @@ $values | ForEach-Object {
...
+#& $env:SystemRoot\\System32\\Sysprep\\Sysprep.exe /oobe /generalize /mode:vm /quit
```
Comment (jadarsie) [inline vhd/packer/sysprep.ps1:31]: does image builder run `sysprep`?

File: temp/AKSeImageTemplateWindows.json (hunk 1 above)
Comment (haofan-ms) [reply temp/AKSeImageTemplateWindows.json]: In the current pipeline, these variables are also from hardcoded values (https://github.com/Azure/aks-engine-azurestack/blob/master/vhd/packer/windows-2019-vars.json), so maybe okay?

File: temp/AKSeImageTemplateWindows.json (hunk 2 above)
Comment (haofan-ms) [reply temp/AKSeImageTemplateWindows.json]: Yes, need them to ensure configurations apply successfully, we have this in current flow as well (https://github.com/Azure/aks-engine-azurestack/blob/master/vhd/packer/windows-vhd-builder.json#L64C9-L71C11).

File: temp/ImageCreationRole.json (hunk 3 above)
Comment (haofan-ms) [reply temp/ImageCreationRole.json]: Removed

File: temp/ImageTemplateLinux.json (hunk 4 above)
Comment (haofan-ms) [reply temp/ImageTemplateLinux.json]: Removed

File: vhd/packer/install-dependencies.sh (hunk 5 above)
Comment (haofan-ms) [reply vhd/packer/install-dependencies.sh:413]: When we call this script from the AIB template, we will ensure 4 arguments are passed into this parameter.

File: vhd/packer/sysprep.ps1 (hunk 6 above)
Comment (haofan-ms) [reply vhd/packer/sysprep.ps1]: This will not be needed, Image Builder service will cover this part

File: vhd/packer/sysprep.ps1 (hunk 7 above)
Comment (haofan-ms) [reply vhd/packer/sysprep.ps1]: Yes it will run sysprep for Windows image

File: packer.mk [hunk 8]
```diff
@@ -19,6 +19,9 @@ run-packer-windows: az-login
 az-copy: az-login
 	azcopy-preview copy "${OS_DISK_SAS}" "${SA_CONTAINER_URL}?${SA_TOKEN}" --overwrite=false
 
+new-az-copy:
```
Comment (jadarsie) [inline packer.mk:22]: could you please rename to `az-copy-rbac`? "new" does not explain what's going on

Comment (haofan-ms) [reply packer.mk]: Updated

**Review by jadarsie — APPROVED** [review]

**Pull request merged**